### PR TITLE
Page data test for InstructorComments

### PR DESCRIPTION
**Pull request by JosephineKwa**

### CODE REVIEW

File: src/main/java/teammates/ui/controller/InstructorCommentsPageData.java [hunk 1]
```diff
@@ -146,6 +120,31 @@ public boolean isViewingDraft() {
...
+                    namesStringBuilder.append("<b>" + student.name + "</b>" + ", ");
+                }
+            } else {
+                namesStringBuilder.append("<b>" + recipient + "</b>" + ", ");
+            }
+            i++;
+        }
+        String namesString = namesStringBuilder.toString();
+        return removeEndComma(namesString);
+    }
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/InstructorCommentsPageData.java]: It is better if this method takes the roster as a parameter. When possible, feed everything the method needs as parameters. That way they are easier to test.

File: src/main/java/teammates/ui/controller/InstructorCommentsPageData.java [hunk 2]
```diff
@@ -146,6 +120,31 @@ public boolean isViewingDraft() {
         return isViewingDraft;
     }        
 
+    private String getRecipientNames(Set<String> recipients) {
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/InstructorCommentsPageData.java]: Can you add a header comment for this method? It's not clear what exactly the method is supposed to do.

File: src/main/java/teammates/ui/controller/InstructorCommentsPageData.java [hunk 3]
```diff
@@ -146,6 +120,31 @@ public boolean isViewingDraft() {
...
+        StringBuilder namesStringBuilder = new StringBuilder();
+        int i = 0;
+        for (String recipient : recipients) {
+            if (i == recipients.size() - 1 && recipients.size() > 1) {
+                namesStringBuilder.append("and ");
+            }
+            StudentAttributes student = roster.getStudentForEmail(recipient);
+            if (courseId.equals(recipient)) {
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/InstructorCommentsPageData.java]: This comparison is not self explanatory. Why compare a course ID to a recipient?

**Comment by JosephineKwa** [comment]
Prof @damithc , regarding the `getRecipientNames` method above, I obtained this method from the actual `*InstructorCommentsPageData` class, as I wanted to use it to programatically create the expected data structures to be compared against the actual data structures created by the init method. However, I will not be using this method in my page data test anymore as I decided to manually create the expected data structures instead.

**Comment by JosephineKwa** [comment]
@bamboo3250 , hope you don't mind reviewing this :)

**Comment by bamboo3250** [comment]
okay, no problem :+1:

File: src/test/java/teammates/test/cases/ui/pagedata/InstructorCommentsPageDataTest.java [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        expectedGivers.addAll(expectedComments.keySet());
+        assertTrue(TestHelper.isSameContentIgnoreOrder(expectedGivers, actualGivers));
+        
+        for (String email : expectedGivers) {
+            assertEquals(expectedComments.get(email), actualComments.get(email));
+        }
+
+        assertEquals(courseId, data.getCourseId());
+        assertEquals(courseName, data.getCourseName());
+        assertTrue(coursePaginationList.equals(data.getCoursePaginationList()));
```
Comment (bamboo3250) [inline src/test/java/teammates/test/cases/ui/pagedata/InstructorCommentsPageDataTest.java]: Use assertEquals here

File: src/test/java/teammates/test/cases/ui/pagedata/InstructorCommentsPageDataTest.java [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+        List<StudentAttributes> studentsInCourse = new ArrayList<StudentAttributes>();
+        for (StudentAttributes student : dataBundle.students.values()) {
+            if (student.course.equals(courseId)) {
+                studentsInCourse.add(student);
+            }
+        }
+        return studentsInCourse;
+    }
+    
+    private void isCommentsForStudentsTablesEqual(CommentsForStudentsTable expected, CommentsForStudentsTable actual) {
```
Comment (bamboo3250) [inline src/test/java/teammates/test/cases/ui/pagedata/InstructorCommentsPageDataTest.java]: Maybe `checkCommentsForStudentsTablesEqual` can be a better name? `is.....` is only for returning `boolean` value.

File: src/test/java/teammates/test/cases/ui/pagedata/InstructorCommentsPageDataTest.java [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+        List<Comment> actualCommentRows = actual.getRows();
+        assertEquals(expectedCommentRows.size(), actualCommentRows.size());
+        for(int i = 0; i < expectedCommentRows.size(); i++) {
+            Comment expectedCommentRow = expectedCommentRows.get(i);
+            Comment actualCommentRow = actualCommentRows.get(i);
+            isCommentRowsEqual(expectedCommentRow, actualCommentRow);
+        }
+    }
+
+    private void isCommentRowsEqual(Comment expected, Comment actual) {
```
Comment (bamboo3250) [inline src/test/java/teammates/test/cases/ui/pagedata/InstructorCommentsPageDataTest.java]: same for this function

**Comment by bamboo3250** [comment]
After finishing those above, feel free to merge it.

**Pull request merged**